testsuite/preview/expansion/src/test/java/org/wildfly/test/preview/microprofile/telemetry/AbstractOpenTelemetryIntegrationTest.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.wildfly.test.preview.microprofile.telemetry;
+
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.core.Response;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.arquillian.testcontainers.api.Testcontainer;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.test.shared.CdiUtils;
+import org.jboss.as.test.shared.observability.containers.OpenTelemetryCollectorContainer;
+import org.jboss.as.test.shared.observability.signals.jaeger.JaegerResponse;
+import org.jboss.dmr.ModelNode;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+
+import java.net.URL;
+import java.util.stream.Collectors;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OUTCOME;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.RESULT;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUBSYSTEM;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUCCESS;
+import static org.jboss.as.test.shared.observability.setuptasks.ServiceNameSetupTask.SERVICE_NAME;
+
+public abstract class AbstractOpenTelemetryIntegrationTest {
+
+    @Testcontainer
+    protected OpenTelemetryCollectorContainer otelCollector;
+
+    @ArquillianResource
+    protected ManagementClient managementClient;
+
+    @ArquillianResource
+    private URL url;
+
+    private static final String MP_CONFIG = "otel.sdk.disabled=false\n" +
+            "otel.metric.export.interval=100";
+
+    static WebArchive createDeployment(String deploymentName) {
+        return ShrinkWrap
+                .create(WebArchive.class, deploymentName + ".war")
+                .addPackage(AbstractOpenTelemetryIntegrationTest.class.getPackage())
```

Comment:
All the tests extending the AbstractOpenTelemetryIntegrationTest are run as client, so do not pollute the deployment with the test class itself.

Reply:
Thank you @rhusar , updated to only package the REST resources in the deployment.